src/lib/storage/fitted_attribute_vector.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <limits>
+#include <memory>
+#include <vector>
+
+#include "base_attribute_vector.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+// An AttributeVector that can adopt to the number of items in the dictionary
+template <typename T>
+class FittedAttributeVector : public BaseAttributeVector {
+ public:
+  explicit FittedAttributeVector(size_t size) { _attribute_vector.resize(size); }
+  // returns the value id at a given position
+  ValueID get(const size_t i) const { return ValueID{_attribute_vector[i]}; }
```

Comment:
Hat es einen Grund, dass die alle in der hpp stehen?

Reply:
Template-Klasse